wifi/views.py

```diff
@@ -0,0 +1,135 @@
+from django.views.generic import View
+from django.core.urlresolvers import reverse
+from notices.decorators import student_profile_complete, default_password_change
+from django.utils.decorators import method_decorator
+from django.contrib import messages
+
+
+class StudentWifiForm(LoginRequiredMixin, View):
+
+    @method_decorator(default_password_change)
+    @method_decorator(student_profile_complete)
+    def get(self, request):
+        user = User.objects.get(username=request.user.username)
+        details = StudentDetail.objects.get(user=user)
+        return render(request, 'wifi/studentwifiform.html', {"user": user, "details": details})
+
+    def post(self, request):
+        hosteler = request.POST.get("hosteler")
+        laptop_mac_address = request.POST.get("laptop_mac_address")
+        user = User.objects.get(username=request.user.username)
+        details = StudentDetail.objects.get(user=user)
+        profile = WifiDetail.objects.filter(user=user)
+        if profile:
+            messages.error(request, "Already Registered")
+            return HttpResponseRedirect(reverse("relevent-notice-list"))
+
+        else:
+            WifiDetail.objects.create(user=user, laptop_mac_address=laptop_mac_address)
+            messages.success(request, "Successfully Registered for Wi-Fi")
+            return HttpResponseRedirect(reverse("relevent-notice-list"))
+
+
+class FacultyWifiForm(LoginRequiredMixin, View):
+
+    @method_decorator(default_password_change)
+    @method_decorator(student_profile_complete)
+    def get(self, request):
+        user = User.objects.get(username=request.user.username)
+        details = FacultyDetail.objects.get(user=user)
+        return render(request, 'wifi/facultywifiform.html', {"user": user, "details": details})
+
+    def post(self, request):
+        laptop_mac_address = request.POST.get("laptop_mac_address")
+        user = User.objects.get(username=request.user.username)
+        details = FacultyDetail.objects.get(user=user)
+        profile = WifiDetail.objects.filter(user=user)
+        if profile:
+            messages.error(request, "Already Registered")
+            return HttpResponseRedirect(reverse("relevent-notice-list"))
+        else:
+            WifiDetail.objects.create(user=user, laptop_mac_address=laptop_mac_address)
+            messages.success(request, "Successfully Registered for Wi-Fi")
+            return HttpResponseRedirect(reverse("relevent-notice-list"))
+
+
+import xlsxwriter
+
+def excel_writer(request):
+    workbook = xlsxwriter.Workbook('Wifi.xlsx')
+    worksheet = workbook.add_worksheet()
+
+    user=User.objects.get(username=request.user.username)
+    details_student = StudentDetail.objects.get(user=user)
+    details_faculty = FacultyDetail.objects.get(user=user)
+    wifi_detail = WifiDetail.objects.get(user=user)
+    length = wifi_detail.count()
+
+    data = ()
+    for i in xrange(len(length)):
+        pass
```

Comment:
@DESHRAJ That I was writing code but I needed to push for @sidjain2901 to work . so I had done that. Well, I will surely remove that sir.